src/khepri_machine.erl

```diff
@@ -166,10 +175,11 @@
          %% Added in machine version 1.
          dedups = #{} :: khepri_machine:dedups_map()}).
 
--opaque state_v1() :: #khepri_machine{}.
-%% State of this Ra state machine, version 1.
+-opaque state_v1_v2() :: #khepri_machine{}.
```

Comment:
The dialyzer seems to be stuck if we write this as:

```erlang
-opaque state_v1() :: #khepri_machine{}.
-opaque state_v2() :: #khepri_machine{}.
-type state() :: state_v2() | state_v1() | khepri_machine_v0:state().
```

so as a compromise I've renamed the `state_v1` type into `state_v1_v2`

Reply:
It would be more verbose, but I believe more correct if there was `tree_v0()` and `tree_v1()` that reference each the correct reverse index. For instance:

```erlang
% In khepri_tree.erl
-type tree_v0() :: #tree{keep_while_conds_revidx ::
                         khepri_tree:keep_while_conds_revidx_v0()}.
-type tree_v1() :: #tree{keep_while_conds_revidx ::
                         khepri_tree:keep_while_conds_revidx_v1()}.
-type tree() :: tree_v0() | tree_v1().
```

Then:

```erlang
% In khepri_machine.erl
-opaque state_v1() :: #khepri_machine{tree :: khepri_tree:tree_v0()}.
-opaque state_v2() :: #khepri_machine{tree :: khepri_tree:tree_v1()}.

-type state() :: state_v2() | state_v1() | khepri_machine_v0:state().
```

What do you think?

Reply:
This is clearer to me :point_up: so I've made the change.

It turns out that the issue with the dialyzer seems to be having a union of opaque types, so if we switch `state_v1` and `state_v2` to be non-opaque, the dialyzer doesn't loop infinitely.